lib/oli_web/live/delivery/student/index_live.ex

```diff
@@ -522,7 +522,11 @@ defmodule OliWeb.Delivery.Student.IndexLive do
       <div class="pr-2 pl-1 self-end">
         <div class="flex items-end gap-1">
           <div class="text-right dark:text-white text-opacity-90 text-xs font-semibold">
-            <%= Utils.days_difference(@lesson.end_date, @ctx) %>
+            <%= case @lesson do %>
+              <% %{end_date: nil} -> %>
+              <% %{end_date: end_date} -> %>
+                <%= Utils.days_difference(end_date, @ctx) %>
```

Comment:
This function is also being used in the [ScheduleComponent](https://github.com/Simon-Initiative/oli-torus/blob/hotfix-v0.28.4/lib/oli_web/live/delivery/student/home/components/schedule_component.ex#L292), so we should fix it there too.
We could also make the fix only in one place, and update the function to return `nil` if the `end_date` is `nil`

Reply:
Good catch!